### PR TITLE
Fix database connection

### LINKED ISSUE #218: Database user authentication when trying to build

### PR DESCRIPTION
**Pull request by shiryz**
Fixes #218 .

@emilyb7 , can you try to run `node server/database/db_build.js`?

### CODE REVIEW

**Comment by emilyb7** [comment]
```
/Users/emilybertwistle/FAC/goals-app-2/goals-app/server/database/db_build.js:11
  if (err) throw err;
           ^

error: role "lrepwxiatobmre" is not permitted to log in
```

:(

**Comment by emilyb7** [comment]
woop i can connect! we have our first user (it's me)

File: server/database/db_build.js [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
-  console.log('Built database with result: ', result);
+client.connect((err) => {
+  if (err) throw err;
+  client.query(sql, (err, result) => {
+    if (err) throw err;
+
+    // disconnect the client
+    client.end((err) => {
+      if (err) throw err;
+      console.log('YAAAs tables were created with result: ', result);
```
Comment (des-des) [inline server/database/db_build.js]: This log..

File: server/database/db_connection.js [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
 };
 
 options.username = username;
 options.password = password;
 options.ssl = (options.host !== 'localhost');
 
-module.exports = new Pool(options);
+module.exports = {
+  client: client,
+  pool: new Pool(options),
```
Comment (des-des) [inline server/database/db_connection.js]: Is this pool being used?

**Comment by shiryz** [comment]
PR might be closed since the code was refactored (again) and another PR #234 is open.

**Pull request closed without merging**